### PR TITLE
Update metrics docs to reflect changes in new version of ASR snippets

### PR DESCRIPTION
**Pull request by k88hudson**
I've modified some of our existing documentation to be more snippets specific and retained the parts that are still relevant, let me know what you think @glogiotatidis.

@ncloudioj can you also take a look at this for accuracy?

### CODE REVIEW

File: docs/data_collection.rst [hunk 1]
```diff
@@ -121,49 +132,14 @@ An impression is whenever a user is shown a specific snippet.
...
-
-Default Browser
-~~~~~~~~~~~~~~~
-
-Some snippets trigger an event that tracks whether Firefox is the default
-browser on the user's system. These snippets also trigger an event when the user
-makes Firefox their default browser by either clicking a link in the snippet or
-by setting the default outside of the browser.
-
-Browser UI Events
```
Comment (k88hudson) [inline docs/data_collection.rst]: This is covered in the Clicks section so I thought it wouldn't be necessary, does that sound right?

File: docs/data_collection.rst [hunk 2]
```diff
@@ -77,41 +77,52 @@ with information about the user's browser.
...
+`Transport Layer Security <https://en.wikipedia.org/wiki/Transport_Layer_Security>`_ to
+securely transmit data to Mozilla servers.
+
+Data collected about snippets is retained on Mozilla secured servers for a period of
+30 days before being rolled up into an anonymous aggregated format.  After this period
+the raw data is deleted permanently. Mozilla **never shares data with any third party**.
+
+The following is an overview of the different kinds of data we collect about snippets:
+
+Message ID
```
Comment (k88hudson) [inline docs/data_collection.rst]: Do we need to include client_id etc. here? What do you think?

Comment (ncloudioj) [reply docs/data_collection.rst]: I don't think so. Instead, we can add a link to [Firefox Privacy Notice](https://www.mozilla.org/en-US/privacy/firefox/), which already covers data collection for Snippets in the "Suggested relevant content" section.

File: docs/data_collection.rst [hunk 3]
```diff
@@ -77,41 +77,52 @@ with information about the user's browser.
...
-   The locale of the current Firefox instance (the same locale value described
-   in the snippet URL from the previous section).
-Country
+   The locale of the current Firefox instance (the same locale value described in the
+   snippet URL from the previous section).
+Region
    The country code corresponding to the country the user is currently located
-   in. This is determined via the user's IP address and is cached locally within
-   Firefox for 30 days. This value may be empty in cases where we can't retrieve
+   in. This value may be empty in cases where we can't retrieve
```
Comment (ncloudioj) [inline docs/data_collection.rst]: We are still getting the region info from user's IP address.

File: docs/data_collection.rst (hunk 1 above)
Comment (ncloudioj) [reply docs/data_collection.rst]: Yep, makes sense.

File: docs/data_collection.rst (hunk 3 above)
Comment (glogiotatidis) [reply docs/data_collection.rst]: nit: missing `.` at the end.

File: docs/data_collection.rst [hunk 4]
```diff
@@ -121,49 +132,14 @@ An impression is whenever a user is shown a specific snippet.
 Snippet Clicks
 ~~~~~~~~~~~~~~
 
-Whenever a link in a snippet is clicked, we trigger an event that notes which
-particular link was clicked. This includes links that may trigger an action
-besides opening up a new page, such as links that trigger browser menus.
+Whenever a link in a snippet is clicked or a form is submitted, we trigger an event
+that notes which particular link was clicked. This includes links that may trigger
+an action besides opening up a new page, such as links that trigger browser menus,
+as well as buttons that submit a form.
```
Comment (glogiotatidis) [inline docs/data_collection.rst]: Can we make this more generic to include all interactions with snippets, like moving from scene 1 to scene 2 on some templates.

**Comment by glogiotatidis** [comment]
@k88hudson can you please update?

**Comment by k88hudson** [comment]
@glogiotatidis hey sorry I did push an update for this – I think I covered all the comments, let me know if there's anything else

**Comment by glogiotatidis** [comment]
Perfect, thanks @k88hudson

**Pull request merged**